base/reflection.jl

```diff
@@ -126,7 +126,12 @@ julia> fieldname(SparseMatrixCSC, 5)
 :nzval
 ```
 """
-fieldname(t::DataType, i::Integer) = t.name.names[i]::Symbol
+function fieldname(t::DataType, i::Integer)
+    n_fields = length(t.name.names)
+    field_label = n_fields == 1 ? "field" : "fields"
+    i > n_fields && error("Cannot access field $i since type $t only has $n_fields $field_label.")
```

Comment:
put the actual error throw into a separate `noline` function to avoid the gc frame/string construction taking up space in the fast path?

Reply:
`|| i < 1`. separate function isn't necessary (no performance concerns for this function)